span_test.go

```diff
@@ -126,6 +126,20 @@ func TestSetTag_SamplingPriority(t *testing.T) {
 	assert.False(t, sp1.context.IsDebug(), "debug should not be allowed by the throttler")
 }
 
+func TestSetTag_SamplingPriority_WithoutDebug(t *testing.T) {
```

Comment:
Could we reduce duplication between this test and the previous test by making them data driven tests? 

I think we should cover the following test cases: 
1. SampleWithoutDebug: false DebugThrottler: default
2. SampleWithoutDebug: true  DebugThrottler: default
3. SampleWithoutDebug: false DebugThrottler: denyAll
3. SampleWithoutDebug: true DebugThrottler: denyAll